packages/bygger/server/backend/Backend.js

```diff
@@ -0,0 +1,176 @@
+    return await Promise.all(
+      forms.map((form) => {
+        return fetchWithErrorHandling(`${updateFormUrl}/${form._id}`, {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+            "x-jwt-token": userToken,
+          },
+          body: JSON.stringify(form),
+        }).then((migratedForm) => migratedForm.data);
+      })
+    );
+  }
+
+  areEqualWithoutWhiteSpaces(string1, string2) {
+    return string1.replace(/\s/g, "") === string2.replace(/\s/g, "");
+  }
+
+  async pushJsonFileToRepo(branch, path, message, file) {
+    const content = stringTobase64(JSON.stringify(file), "utf-8");
+    const remoteFile = await this.skjemaUtfylling.getFileIfItExists(branch, path);
+    const sha = remoteFile && remoteFile.data && remoteFile.data.sha;
+
+    if (
+      remoteFile &&
+      remoteFile.data &&
+      remoteFile.data.content &&
+      this.areEqualWithoutWhiteSpaces(remoteFile.data.content, content)
+    ) {
+      // The file exists remotely, and is identical to the "local" file. Skip update.
+      return Promise.resolve(undefined);
+    }
+
+    const createResult = await this.skjemaUtfylling.createOrUpdateFileContents(branch, path, message, content, sha);
+    return createResult.data.commit.sha;
+  }
+
+  async performChangesOnSeparateBranch(base, branch, performChanges) {
+    const baseRef = await this.skjemaUtfylling.getRef(base);
+    await this.skjemaUtfylling.createRef(branch, baseRef.data.object.sha);
+
+    await performChanges(branch);
+
+    let updatedBaseSha;
+    if (await this.skjemaUtfylling.hasBranchChanged(baseRef, branch)) {
+      // Only create and merge pull request if the branch contains changes, compared to the base branch
+      const pullRequest = await this.skjemaUtfylling.createPullRequest("Automatic publishing job", branch, base);
+      await this.skjemaUtfylling.mergePullRequest(pullRequest.data.number);
+      const updatedBase = await this.skjemaUtfylling.getRef(base);
+      updatedBaseSha = updatedBase.data.object.sha;
+    }
+
+    await this.skjemaUtfylling.deleteRef(branch);
+    return updatedBaseSha;
+  }
+
+  pushFilesAndUpdateSubmoduleCallback(files) {
+    return async (branch) => {
+      const initialRef = await this.skjemaUtfylling.getRef(branch);
+
+      for (const file of files) {
+        await this.pushJsonFileToRepo(
+          branch,
+          file.path,
+          `[Publisering] ${file.type} "${file.name}", monorepo ref: ${this.config.gitSha}`,
+          file.content
+        );
+      }
+
+      if (await this.skjemaUtfylling.hasBranchChanged(initialRef, branch)) {
+        await this.skjemaUtfylling.updateSubmodule(
+          branch,
+          this.config.gitSha,
+          this.config.submoduleRepo,
+          `[Publisering] oppdater monorepo ref: ${this.config.gitSha}`
+        );
+      }
+    };
+  }
+
+  async publishForm(userToken, formContent, translationsContent, formPath) {
+    const form = {
```

Comment:
Burde man ha en funksjon for opprette "fil" for å sikre riktige verdier blir satt.